### DISCUSSION TITLE
Add controlled mode to select

### DISCUSSION THREAD

**Discussion by Anon-136** [MobileReality/react-native-select-pro#199] [General]
**Is your feature request related to a problem? Please describe.**
I run to this problem where I want to have a confirm modal popping up after an option is selected.
But after I select the state has changed already and I have no way to control it and change it back if the user choose to cancel it.

**Describe the solution you'd like**
I want to have a control of the internal state: selectedOptionIndex, selectedOption, or else.
A nice react `Select` library should have 2 mode: uncontrolled and controlled.
It can be like
`<Select value={...} options={[...]}>`

**Additional context**
Some hacky solution: I remount the component by setting the new `key` and set default option to be the other one.

**Comment by kkaplinski** [comment]
Hey @Anon-136 , thank you for your comment. Can you try to change it back with `clear` method called on cancel ? https://mobilereality.github.io/react-native-select-pro/docs/api/select#methods.

**Reply by Anon-136** [reply]
I did try but clear will remove the selected option completely.
What i want is to change the select back to the default one

**Reply by irekrog** [reply]
@Anon-136 At this moment we could add a new method `reset` to back to the default option. Would be ok for you?
